### PR TITLE
Closes #126 - Update check for filename to allow underscore

### LINKED ISSUE #126: This check the output file name not the data value in the data frame
https://github.com/atorus-research/xportr/blob/231e959b84aa0f1e71113c85332de33a827e650a/R/write.R#LL33C1-L33C82
Could you please confrim this check refer to the output file name not the data value?
For the file name, if the DM_F can be used when the `strict_checks = FALSE`，and the DM_F means the full SDTM  that contains the parent DM and SUPPDM
Thank you very much.

### PR DESCRIPTION
**Pull request by cpiraux**
### Thank you for your Pull Request!

We have developed a Pull Request template to aid you and our reviewers. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs. We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

Move the check for the file name to strict_check to allow underscores in the filename.

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Make sure that the pacakge version in the NEWS.md and DESCRIPTION file is same. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/write.R [hunk 1]
```diff
@@ -75,12 +75,12 @@ xportr_write <- function(.df,
     abort("`.df` file name must be 8 characters or less.")
   }
 
+  checks <- xpt_validate(.df)
+
   if (stringr::str_detect(name, "[^a-zA-Z0-9]")) {
```
Comment (bms63) [inline R/write.R:80]: can you make a comment in the code on why this is relaxed

**Review by bms63 — COMMENTED** [review]
This is newsworth!  Can you make a short blurb in the NEws section about this relaxation of our checks

File: R/write.R (hunk 1 above)
Comment (cpiraux) [reply R/write.R:80]: Great suggestion! I've made an update to New.md by adding a brief description.

**Review by bms63 — APPROVED** [review]
Let's go!!  Thanks for doing this one!!

**Pull request merged**